ydb/library/yql/sql/settings/translation_settings.cpp

```diff
@@ -67,6 +67,37 @@ namespace NSQLTranslation {
         , AssumeYdbOnClusterWithSlash(false)
     {}
 
+    NYql::NProto::TTranslationSettings TTranslationSettings::Serialize() const {
+        NYql::NProto::TTranslationSettings serializedSettings;
+
+        serializedSettings.SetPathPrefix(PathPrefix);
+        serializedSettings.SetSyntaxVersion(SyntaxVersion);
+        serializedSettings.SetAnsiLexer(AnsiLexer);
+        serializedSettings.SetAntlr4Parser(Antlr4Parser);
+        serializedSettings.SetPgParser(PgParser);
+
+        auto* pragmas = serializedSettings.MutablePragmas();
+        pragmas->Add(Flags.begin(), Flags.end());
+
+        return serializedSettings;
+    }
+
+    void TTranslationSettings::UpdateWith(const NYql::NProto::TTranslationSettings& serializedSettings) {
+        #define DeserializeSetting(settingName) \
+            if (serializedSettings.Has##settingName()) { \
+                settingName = serializedSettings.Get##settingName(); \
+            }
+
+        DeserializeSetting(PathPrefix);
+        DeserializeSetting(SyntaxVersion);
+        DeserializeSetting(AnsiLexer);
+        DeserializeSetting(Antlr4Parser);
+        DeserializeSetting(PgParser);
+
+        #undef DeserializeSetting
+
+        Flags.insert(serializedSettings.GetPragmas().begin(), serializedSettings.GetPragmas().end());
```

Comment:
#### What if the `Flags` already contain [`AnsiCurrentRow`](https://ydb.tech/docs/en/yql/reference/syntax/pragma#ansicurrentrow) and `serializedSettings` are adding `DisableAnsiCurrentRow`?

See the relevant code [here](https://github.com/ydb-platform/ydb/blob/f38e45016276218330480a40a5575b3d5523522e/ydb/library/yql/sql/v1/context.cpp#L112). I believe, the last pragma will win, i.e. `DisableAnsiCurrentRow` will be in effect.

Takeaway: serialized settings will take precedence over those fields of the TTranslationSettings struct that they are updating.